### DISCUSSION TITLE
invalid args `order` for command `add_order`: trailing input

### DISCUSSION THREAD

**Discussion by b2nil** [tauri-apps/tauri#5821] [Q&A] [answered]
Hi,

I've used vue + vuetifyjs as the frontend framework, and met this error when submitting an order form from the frontend.
> invalid args `order` for command `add_order`: trailing input

This error message is so confusing and cannot be debugged, at least for me.
Anyone had similiar experience?

Here below is my tauri command `add_order`: 
```rust
#[tauri::command]
pub fn add_order(order: NewOrder) -> usize {
    let mut conn = establish_conn();
    let res = diesel::insert_or_ignore_into(schema::orders::dsl::orders)
        .values(order)
        .execute(&mut conn)
        .map_err(|e| {
            println!("{}", format!("error on [add_order]: {:?}", e));
        })
        .unwrap_or(0);

    return res;
}
```

And at the frontend, I invoked the tauri command like below:
```typescript
export async function addOrder(order: Partial<Order>) {
    return invoke<number>("add_order", { order })
}
```
And this is the type I passed to addOrder. (Note that backend order type does not have id field.)
```typescript
export interface Order {
    id?: number
    po_num: string,
    start_date: Date,
    finish_date: Date,
    job_type: string,
    description: string,
    customer_name: string,
    customer_code: string,
    currency: string,
    exchange_rate: number,
    qty: number,
    rate: number,
    amount: number,
}
```

**Comment by FabianLars** [comment, marked as answer]
The issue should be the `Date` type. I saw this in my own app (without tauri) once, and another Tauri user reported the same issue (with Date also being the problem).

One related issue: [serde-rs/json#291](https://togithub.com/serde-rs/json/issues/291)

**Reply by b2nil** [reply]
Thank you so much for the help.
